docs/source/extension/extension_points.rst

```diff
@@ -721,6 +723,117 @@ When the ``labStatus`` busy state changes, we update the text content of the
 
 .. _widget-tracker:
 
+
+Toolbar Registry
+----------------
+
+As for the context menu and the main menu bar, JupyterLab provides an infrastructure to define
+and customize toolbar widgets of document widgets from the settings. A typical example is
+the notebook toolbar as seen in the snippet below:
+
+.. code:: typescript
+
+   function activatePlugin(
+     app: JupyterFrontEnd,
+     // ...
+     toolbarRegistry: IToolbarWidgetRegistry | null,
+     settingRegistry: ISettingRegistry | null
+   ): NotebookWidgetFactory.IFactory {
+     const { commands } = app;
+     let toolbarFactory:
+       | ((widget: NotebookPanel) => DocumentRegistry.IToolbarItem[])
+       | undefined;
+   
+     // Register notebook toolbar widgets
+     if (toolbarRegistry) {
+       toolbarRegistry.registerFactory<NotebookPanel>(FACTORY, 'cellType', panel =>
+         ToolbarItems.createCellTypeItem(panel, translator)
+       );
+       
+       toolbarRegistry.registerFactory<NotebookPanel>(
+         FACTORY,
+         'kernelStatus',
+         panel => Toolbar.createKernelStatusItem(panel.sessionContext, translator)
+       );
+       // etc... 
+     
+       if (settingRegistry) {
+         // Create the factory
+         toolbarFactory = createToolbarFactory(
+           toolbarRegistry,
+           settingRegistry,
+           // Factory name
+           FACTORY,
+           // Setting id in which the toolbar items are defined
+           '@jupyterlab/notebook-extension:panel',
+           translator
+         );
+       }
+     }
+   
+     const factory = new NotebookWidgetFactory({
+       name: FACTORY,
+       fileTypes: ['notebook'],
+       modelName: 'notebook',
+       defaultFor: ['notebook'],
+       // ...
+       toolbarFactory,
+       translator: translator
+     });
+     app.docRegistry.addWidgetFactory(factory);
+
+The registry ``registerFactory`` method allows an extension to provide special widget for a unique pair 
+(factory name, toolbar item name). Then the helper ``createToolbarFactory`` can be used to extract the
+toolbar definition from the settings and build the factory to pass to the widget factory.
+
+The default toolbar items can be defined across multiple extensions by providing an entry in the ``"jupyter.lab.toolbars"``
+mapping. For example for the notebook panel:
+
+.. code:: json
+ 
+   "jupyter.lab.toolbars": {
+     "Notebook": [
```

Comment:
Is the key (`"Notebook"`)  meaningful, or can it be anything? I would not know after reading the docs only.

Reply:
Indeed it is relevant as this should match the unique widget factory name. I'll update the example.